cni/pkg/ambient/net_linux.go

```diff
@@ -818,21 +818,30 @@ func (s *Server) CreateEBPFRulesWithinNodeProxyNS(proxyNsVethIdx int, ztunnelIP,
 			}
 		}
 
-		// Flush prerouting table - this should be a new pod netns and it should be clean, but just to be safe..
-		err = execute(s.IptablesCmd(), "-t", "mangle", "-F", "PREROUTING")
+		vethLink, err := netlink.LinkByIndex(proxyNsVethIdx)
 		if err != nil {
-			return fmt.Errorf("failed to configure iptables rule: %v", err)
+			return fmt.Errorf("failed to find veth with index '%d' within namespace %s: %v", proxyNsVethIdx, ztunnelNetNS, err)
 		}
 
-		// Logging for prerouting mangle
-		err = execute(s.IptablesCmd(), "-t", "mangle", "-I", "PREROUTING", "-j", "LOG", "--log-prefix", "ztunnel mangle pre")
+		err = disableRPFiltersForLink(vethLink.Attrs().Name)
 		if err != nil {
-			return fmt.Errorf("failed to configure iptables rule: %v", err)
+			log.Warnf("failed to disable procfs rp_filter for device %s: %v", vethLink.Attrs().Name, err)
 		}
 
-		vethLink, err := netlink.LinkByIndex(proxyNsVethIdx)
+		if ebpf.EBPFTproxySupport() {
+			log.Infof("Current kernel version support tproxy in eBPF, skip the iptables rules.")
+			return nil
+		}
```

Comment:
If everything that follows this is an optional block, can we move everything that follows this into a dedicated function that's easier to reason about independently?

Reply:
updated.